src/LayerEditor/ui/diagram_editor/graphics_items/gs_segment.py

```diff
@@ -98,5 +115,10 @@ def itemChange(self, change, value):
 
     def update_zoom(self, value):
         pen = QtGui.QPen()
-        pen.setWidthF(self.WIDTH * 2 / value)
-        self.setPen(pen)
\ No newline at end of file
+        pen.setWidthF(self.WIDTH / value)
+        self.setPen(pen)
+
+
+    def update_color(self, layer_id: int, regions: RegionsModel):
```

Comment:
Same as fo GsPoint
... Possibly derive these from common GsObject?

Reply:
GSPoint, GSSegment and GSPolygon are very similar, so they might benefit from inheriting common base object. I can look into it if you want.